### PR TITLE
Improve docs around pipeline target types

### LINKED ISSUE #3316: Improve docs around pipeline target types
### Description

Follow up to discussion in support-tickets

- [ ] List out the "Target Types" for a pipeline stage
- [ ] Additional info around device groups 
- [ ] Link back to the product page for feature availability (single source of truth)

### Epic/Story

_No response_

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

### PR DESCRIPTION
**Pull request by Steve-Mcl**
closes #3316 

## Description

* List out the "Target Types" for a pipeline stage
* List out the "Actions" for a pipeline stage
* Additional info around device groups
* Add feature availability for device groups

## Related Issue(s)

#3316 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [x] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by ZJvandeWeg** [comment]
@joepavitt can you assist from a user perspective? And we both do a pass?

File: docs/user/device-groups.md
Comment (ZJvandeWeg) [inline docs/user/device-groups.md:11]: Is there another way to deploy to these groups? `can` is implying that for me. In other words, is there tight coupling with devops pipelines or not?

File: docs/user/devops-pipelines.md
Comment (ZJvandeWeg) [inline docs/user/devops-pipelines.md:30]:
```suggestion
2. Click **Duplicate Instance** and provide the necessary details
```

File: docs/user/devops-pipelines.md
Comment (ZJvandeWeg) [inline docs/user/devops-pipelines.md:39]:
```suggestion
See [Pipeline Stage details](#pipeline-stage-details) below for more info.
```

File: docs/user/device-groups.md
Comment (ZJvandeWeg) [inline docs/user/device-groups.md:10]: I believe the _why_ a user wants this isn't well covered.

When managing many devices running the same snapshot, ..... etc etc

File: docs/user/devops-pipelines.md
Comment (ZJvandeWeg) [inline docs/user/devops-pipelines.md:55]: As an engineer I like this section, though difficult to parse. No changes required, I believe almost all specs are

File: docs/user/device-groups.md
Comment (ZJvandeWeg) [inline docs/user/device-groups.md:40]: What happens after? What if I added a device to a group and the group is supposed to run snapshot foobar, does it run foobar too?

File: docs/user/device-groups.md
Comment (ZJvandeWeg) [inline docs/user/device-groups.md:43]:
```suggestion
_Note: When a device you want to add to a group doesn't appear in the list, it's likely already assigned to another group._
```

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

File: docs/user/device-groups.md
Comment (Steve-Mcl) [reply docs/user/device-groups.md:17]: Device groups are Enterprise right? - can you suggest suitable wording please @ZJvandeWeg

File: docs/user/device-groups.md
Comment (Steve-Mcl) [reply docs/user/device-groups.md:21]: You can, if you wish, create a group, then go create devices, then come back and update them, but for the sake of simplicity I added that line - however i completely understand the point you are making. I will revisit / revise.

File: docs/user/device-groups.md
Comment (Steve-Mcl) [reply docs/user/device-groups.md:40]: That will be added in the PR on branch [3260-update-device-on-group-membership-change](https://github.com/FlowFuse/flowfuse/tree/3260-update-device-on-group-membership-change) and why I am holding off raising that PR until this PR is merged.

File: docs/user/device-groups.md
Comment (joepavitt) [reply docs/user/device-groups.md:18]: Running with an EE license is difference to a given Team Tier. FF Cloud runs on an EE License, but only Enterprise Teams have access to Device Groups iirc?

File: docs/user/device-groups.md
Comment (joepavitt) [reply docs/user/device-groups.md]: Yeah, to echo @ZJvandeWeg here - the "ensure you have 2 or more devices" is odd phrasing, surely the minimum is 1, if any at all?

File: docs/user/devops-pipelines.md
Comment (Steve-Mcl) [reply docs/user/devops-pipelines.md:97]:
```suggestion
When a pipeline stage with this action is deployed, the latest existing snapshot of the defined instance will be pushed to the next stage.
```

@joepavitt counter proposal ^

File: docs/user/device-groups.md
Comment (Steve-Mcl) [reply docs/user/device-groups.md]: removed whole sentence (not relevant)

File: docs/user/device-groups.md
Comment (Steve-Mcl) [reply docs/user/device-groups.md:20]: Now updated to:

> The following requirements apply:
> 
>  - FlowFuse 1.15+ Enterprise tier
> - FlowFuse Cloud

Please offer alternative suggestion if not ok

@ZJvandeWeg @joepavitt

File: docs/user/devops-pipelines.md
Comment (Steve-Mcl) [reply docs/user/devops-pipelines.md:57]: I originally had that @joepavitt. I feel it infers that _any_ stage on the left can affect any upstream stage on the right.  My re-write is technically more accurate (if somewhat less succinct).  

Skipping this suggestion unless otherwise contested

File: docs/user/devops-pipelines.md
Comment (Steve-Mcl) [reply docs/user/devops-pipelines.md:97]: I'm going with my counter proposal. It can be contested in the re-review.

**Comment by Steve-Mcl** [comment]
@ZJvandeWeg @joepavitt I beleive I have addressed your many many many comments (yikes - didnt realise I was sooooo bad at docs!)

Can you both give it one last look over please so we can get it merged and I can pull/merge and finish up the PR on branch [3260-update-device-on-group-membership-change](https://github.com/FlowFuse/flowfuse/tree/3260-update-device-on-group-membership-change)

File: docs/user/device-groups.md
Comment (knolleary) [reply docs/user/device-groups.md:20]: @Steve-Mcl a more correct wording is:

> - FlowFuse 1.15+ running with an EE license
> - FlowFuse Cloud Enterprise Teams

Comment (joepavitt) [reply docs/user/device-groups.md:20]: Will leave it to ZJ to work on this, I _believe_ he's asking for "FlowFuse Cloud - Enterprise Team Tier", but I'm not 100% certain

Comment (Steve-Mcl) [reply docs/user/device-groups.md]: Updated to 

> The following requirements apply:
> - FlowFuse 1.15+ running with an EE license
> - FlowFuse Cloud Enterprise Teams

Thanks @knolleary

File: docs/user/devops-pipelines.md
Comment (Steve-Mcl) [reply docs/user/devops-pipelines.md:57]: So I can move forward, I will apply suggestion.

**Comment by Steve-Mcl** [comment]
Final tweaks complete.

File: docs/user/device-groups.md
Comment (ZJvandeWeg) [inline docs/user/device-groups.md:19]: @Steve-Mcl There's no EE license, it doesn't exist. It's the Enterprise tier.

File: docs/user/device-groups.md
Comment (ZJvandeWeg) [inline docs/user/device-groups.md:20]:
```suggestion
- FlowFuse Cloud teams on the Enterprise tier
```

**Comment by ZJvandeWeg** [comment]
Thanks for the additional screenshots!

File: docs/user/device-groups.md
Comment (Steve-Mcl) [reply docs/user/device-groups.md:19]: @ZJvandeWeg this is coflicting with @knolleary suggestion: https://github.com/FlowFuse/flowfuse/pull/3318#discussion_r1449053434

Please advise.

Comment (knolleary) [reply docs/user/device-groups.md:19]: @Steve-Mcl it isn't conflicting - its an improved iteration. 

```suggestion
- FlowFuse 1.15+ Enterprise Tier
```

**Comment by Steve-Mcl** [comment]
@ZJvandeWeg 

final wording around requirements is now:

> The following requirements apply:
> 
> - FlowFuse 1.15+ Enterprise Tier
> - FlowFuse Cloud teams on the Enterprise tier

Does this read how you expect/hope 🤞

**Comment by joepavitt** [comment]
Just pushed some formatting changes, the stage actions being H4's made the article difficult to read, so changed them to boldened bullets instead.

**Review by ZJvandeWeg — APPROVED** [review]

**Pull request merged**
